### ISSUE TITLE
@RelationshipLinks also serialized to attributes

### ISSUE THREAD

**Issue by marceloverdijk** [jasminb/jsonapi-converter#165]
E.g. taking this resource:

```
@Type("person")
public class PersonResource extends BaseResource {

    private String name;
    .. other person properties left out out for simplicity ..

    @Relationship("homeworld")
    private PlanetResource homeworld;

    @RelationshipLinks("homeworld")
    private Links homeworldLinks;
```

is serialized to:

```
{
    "data": [
        {
            "links": {
                "self": {
                    "href": "http://localhost:8077/persons/27"
                }
            },
            "type": "person",
            "id": "27",
            "attributes": {
                "name": "Ackbar",
                "height": 180,
                "mass": 83,
                "skin_color": "brown mottle",
                "eye_color": "orange",
                "birth_year": "41BBY",
                "gender": "male",
                "homeworld_links": {
                    "links": {
                        "self": {
                            "href": "http://localhost:8077/planets/31"
                        }
                    }
                }
            },
            "relationships": {
                "homeworld": {
                    "links": {
                        "self": {
                            "href": "http://localhost:8077/planets/31"
                        }
                    },
                    "data": {
                        "type": "planet",
                        "id": "31"
                    }
                }
            }
        },
        {
          ..
        },
        ..
```

As you can see the relationships nicely contain the links and the data.

But the `attributes` now also contains a `homeworld_links` which is not correct.

I can easily working around it by adding a `@JsonIgnore` to the `homeworldLinks` like:

```
    @RelationshipLinks("homeworld")
    @JsonIgnore
    private Links homeworldLinks;
```

I think the lib should do this automatically or am I doing something wrong?

Note the `homeworld` property is not part of the `attributes` as expected; not needing an explicit `@JsonIgnote` annotation.

**Comment by marceloverdijk** [comment]
The problem is here: https://github.com/jasminb/jsonapi-converter/blob/develop/src/main/java/com/github/jasminb/jsonapi/ResourceConverter.java#L857

In my case the `attributesNode` contains the field `homeworld_links ` (as my object mapper is configured with snake case).

But the mentioned line is trying to remove `homeworldLinks` as that is the name of Java bean property it found.

**Comment by marceloverdijk** [comment]
Closed as #166 is merged.
